src/Entity/CertificationRequest.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use Doctrine\ORM\Mapping as ORM;
+use Ramsey\Uuid\Uuid;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\CertificationRequestRepository")
+ * @ORM\Table(name="certification_request")
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class CertificationRequest
+{
+    use EntityIdentityTrait;
+    use EntityTimestampableTrait;
+
+    public const STATUS_PENDING = 'pending';
+    public const STATUS_APPROVED = 'approved';
+    public const STATUS_REFUSED = 'refused';
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=20)
+     */
+    private $status;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(nullable=true)
+     */
+    private $documentName;
+
+    /**
+     * @var UploadedFile|null
+     *
+     * @Assert\File(
+     *     maxSize="5M",
+     *     mimeTypes={
+     *         "application/pdf",
+     *         "application/x-pdf",
+     *         "image/jpeg",
+     *         "image/png"
+     *     },
+     *     mimeTypesMessage="certification_request.document.mime_type"
+     * )
+     */
+    private $document;
+
+    /**
+     * @var Adherent
+     *
+     * @ORM\ManyToOne(targetEntity=Adherent::class, inversedBy="certificationRequests")
+     * @ORM\JoinColumn(nullable=false, onDelete="CASCADE")
+     */
+    private $adherent;
+
+    /**
+     * @var Administrator|null
+     *
+     * @ORM\ManyToOne(targetEntity=Administrator::class)
+     * @ORM\JoinColumn(nullable=true, onDelete="SET NULL")
+     */
+    private $processedBy;
+
+    public function __construct(Adherent $adherent)
+    {
+        $this->adherent = $adherent;
+        $this->uuid = Uuid::uuid4();
+        $this->status = self::STATUS_PENDING;
+    }
+
+    public function __toString()
+    {
+        return (string) $this->adherent;
+    }
+
+    public function getStatus(): string
+    {
+        return $this->status;
+    }
+
+    public function getDocumentName(): ?string
+    {
+        return $this->documentName;
+    }
+
+    public function getDocumentExtension(): ?string
```

Comment:
It will be nice to save mimeType directly from `UploadedFile::getMimeType()` for reusing it with `<embed mimetype="..." />` tag on twig, wdyt?